### ISSUE TITLE
FileUpload: First multiple upload after fresh deploy fails on a random number of files when `validateContenType=true`

### ISSUE THREAD

**Issue by gmanjon** [primefaces/primefaces#8797]
### Describe the bug

After a fresh deployment, when multiple files are being uploaded in the very first upload, a small random number of them fail to reach the fileuplaod component listener method, and the server ouputs the following log:

> Could not determine content type of uploaded file _filename_here_, consider plugging in an adequate FileTypeDetector implementation

The log appears the exact same number of times as the number of files that doesn't reach the fileupload component listener method.

After that operation, all uploads work properly until the next restart (redeployment or Application Server restart).

This doesn't happen if just one file is uploaded.

This happens only if `validateContenType="true"` 

This happens with both Tika and mime-types providers.

**Cause Theory**

* `PrimeApplicationContext` is a Singleton, and hence, its `FileTypeDetector fileTypeDetector` is shared between different threads. 
* Fileupload component sends each file in a different request, making them to arrive concurrently
* The `FileTypeDetector` (anonymous inner class in `PrimeApplicationContext`) uses an instance of `ServiceLoader<FileTypeDetector>` to resolve the service providers.
* According to the ServiceLoader official documentation (https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/ServiceLoader.html): _Instances of this class are not safe for use by multiple concurrent threads._

This means that first services resolution are happening simultaneosly (each sent file) on the same ServiceLoader (one instantiation in a Singleton class), and probably this is causing trouble in this very moment, the first resolution in the first multiple upload.

This would explain why in doesn't happen when uplaoding only one file the first time, and why after the first upload all other uplaods work properly, because the Service has already been resolved.

### Reproducer

I added two files

- `primefaces-test` contains the test with the primefaces template from its repository. With this one, I don't know exactly why (but I presume is related to some kind of performance difference in ServiceLoader between Jetty and Wildfly), I could only reproduce it once. It didn't happen in any other try (aprocimately 10 times I tried)
- `primefaces-bug` contains a minimal reproducible example without the template, I use it to deploy it ina WIldfly 20.0.1.Final, and there it reproduces the error each and every time.

In any of them, to try to reproduce it, just:

1. Open homepage
2. Click upload files
3. Select a large number of files, 10 for instance.
4. Click Upload
5. The listener method will output the log for some of them, and for the rest a `Could not determine content type of uploaded file _filename_here_, consider plugging in an adequate FileTypeDetector implementation` message will be shown in the server output.

[primefaces-bug.zip](https://github.com/primefaces/primefaces/files/8721787/primefaces-bug.zip)
[primefaces-test.zip](https://github.com/primefaces/primefaces/files/8721788/primefaces-test.zip)

### Expected behavior

Any multiple fileupload should work properly, even the first uploads after a fresh deploy.

### PrimeFaces edition

Community

### PrimeFaces version

tested on 10.0.0 and 11.0.0

### Theme

default

### JSF implementation

Mojarra

### JSF version

2.3

### Browser(s)

Chrome Version 100.0.4896.127 (Official Build) (64-bit)

**Comment by tandraschko** [comment]
can you please try again?

**Comment by gmanjon** [comment]
Hi @tandraschko,

That was fast, thank you very much.

I downloaded master an tested locally successfully. Everything works fine now. I tried many times an none failed, and as soon as switched to 11.0.0 it failed again.

I built it from source because I couldnt find any SNAPSHOT neither in sonatype (only 4.1.0-SNAPSHOT there) or in http://repository.primefaces.org (Primefaces 8 is the last one there). Is there some other repo? or SNAPSHOT are always built locally?

On the other hand, there will be a fix for Primefaces 10 about this? If not I'm trying a workaround (as I cannot afford losing uploads in production after every deployment)

I'm trying to do this after Primefaces initialization:

```java
File file = File.createTempFile("filetypedetector", System.currentTimeMillis() + "");
PrimeApplicationContext.getCurrentInstance(facesContext).getFileTypeDetector().probeContentType(file.toPath());
```

The problem I'm having is reaching that point after `PrimeApplicationContext` as been created. With `@Singleton @Startup`, `PrimeApplicationContext` has not been created yet and there is still no `FacesContext` to build it for the first time. The same with `ServletContextListener` or even creating another `Servlet`, the context is totally not there obviously.

Can you think of any other suggestion? Thank you so much.

**Comment by tandraschko** [comment]
check our github index, there we describe how to get the snapshot

also read this for your PF 10 fix: https://github.com/primefaces/primefaces#community--elite--pro


A valid startup listener for JSF is to use a PostConstructApplication event observer, just google a bit

**Comment by gmanjon** [comment]
Yep, all the questions were answered in the main README, sorry to bother you (I know that's quite annoying).

And about the workaround I've been actually googling quite a bit before asking but didn't hit that specific event.  Anyway I'm receiving an exception trying to initialize Primefaces at that stage:

```
Caused by: java.lang.UnsupportedOperationException: UT010042: This method cannot be called from a servlet context listener that has been added programatically
     at io.undertow.servlet@2.1.3.Final//io.undertow.servlet.spec.ServletContextImpl.ensureNotProgramaticListener(ServletContextImpl.java:976)
     at io.undertow.servlet@2.1.3.Final//io.undertow.servlet.spec.ServletContextImpl.getSessionCookieConfig(ServletContextImpl.java:676)
     at io.undertow.servlet@2.1.3.Final//io.undertow.servlet.spec.ServletContextImpl.getSessionCookieConfig(ServletContextImpl.java:110)
     at deployment.xxx.war//org.primefaces.config.PrimeConfiguration.<init>(PrimeConfiguration.java:155)
     at deployment.xxx.war//org.primefaces.context.PrimeApplicationContext.<init>(PrimeApplicationContext.java:88)
     at deployment.xxx.war//org.primefaces.context.PrimeApplicationContext.getCurrentInstance(PrimeApplicationContext.java:221)
     at deployment.xxx.war//org.primefaces.context.PrimeApplicationContext.getCurrentInstance(PrimeApplicationContext.java:209)
```

No worries, I will use a custom 10.0.0 fixed untill 12.0.0 is realeased and I have time to upgrade.

Thank you for the help, and sorry again for asking things already stated in the README.
